src/main/java/org/sonatype/nexus/plugins/conda/internal/hosted/CondaHostedRecipe.groovy

```diff
@@ -0,0 +1,164 @@
+import org.sonatype.nexus.repository.view.matchers.ActionMatcher
+
+import javax.annotation.Nonnull
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton
+
+import static org.sonatype.nexus.repository.http.HttpMethods.*
+import static org.sonatype.nexus.repository.http.HttpResponses.methodNotAllowed
+import static org.sonatype.nexus.repository.http.HttpResponses.notFound
+import static org.sonatype.nexus.repository.http.HttpResponses.ok
+import static org.sonatype.nexus.repository.view.matchers.logic.LogicMatchers.and
+
+/**
+ * Conda Hosted Recipe
+ */
+@Named(CondaHostedRecipe.NAME)
+@Singleton
+class CondaHostedRecipe
+        extends CondaRecipeSupport {
+    public static final String NAME = 'conda-hosted'
+
+    @Inject
+    Provider<CondaHostedFacetImpl> hostedFacet
+
+    @Inject
+    CondaHostedRecipe(@Named(HostedType.NAME) final Type type, @Named(CondaFormat.NAME) final Format format) {
+        super(type, format)
+    }
+
+    @Override
+    void apply(@Nonnull final Repository repository) throws Exception {
+        repository.attach(securityFacet.get())
+        repository.attach(configure(viewFacet.get()))
+        repository.attach(httpClientFacet.get())
+        repository.attach(componentMaintenanceFacet.get())
+        repository.attach(storageFacet.get())
+        repository.attach(hostedFacet.get())
+        repository.attach(searchFacet.get())
+        repository.attach(attributesFacet.get())
+    }
+
+
+    /**
+     * Match conda metadata files
+     */
+    static Matcher condaMetaDataMather = {
+        Context context ->
+            def path = context.getRequest().getPath()
+            log.warn("Searching condaMetaDataMather " + path)
+            String candidate = path.substring(path.lastIndexOf('/'))
+            CondaPathUtils.CONDA_META_DATA.contains(candidate)
+    }
+
+    /**
+     * Conda path matcher - verifies path has conda alike structure
+     */
+    static Matcher condaPathMatcher = {
+        Context context ->
+            // TODO: fix logging, use try monad
+            try {
+                final String path = context.getRequest().getPath()
+                log.warn("Searching condaPathMatcher " + path)
+                CondaPath.build(path)
+            }
+            catch (Exception ex) {
+                false
+            }
+            true
+    }
+
+    static Matcher fetchMetaDataMatcher = and(new ActionMatcher(GET, HEAD), condaMetaDataMather)
+    static Matcher fetchCondaFileMatcher = and(new ActionMatcher(GET, HEAD), condaPathMatcher)
+    static Matcher uploadCondaFileMatcher = and(new ActionMatcher(PUT), condaPathMatcher)
+    static Matcher deleteCondaFileMatcher = and(new ActionMatcher(DELETE), condaPathMatcher)
+
+    static String removeLeadingSlash(String path) {
+        path.length() > 1 && path[0] == '/' ? path.substring(1) : path
+    }
+
+    final Handler handler = new Handler() {
```

Comment:
It's interesting how you've implemented this. I've tended in the past to do something more akin to: https://github.com/sonatype-nexus-community/nexus-repository-helm/blob/master/src/main/java/org/sonatype/repository/helm/internal/hosted/HostedHandlers.java

I would suggest moving them out of the recipe and into their own class, etc...